persist/fs/files.go

```diff
@@ -444,3 +703,8 @@ func filesetPathFromTime(prefix string, t time.Time, suffix string) string {
 	name := fmt.Sprintf("%s%s%d%s%s%s", filesetFilePrefix, separator, t.UnixNano(), separator, suffix, fileSuffix)
 	return path.Join(prefix, name)
 }
+
+func snapshotPathFromTimeAndIndex(prefix string, t time.Time, suffix string, index int) string {
+	name := fmt.Sprintf("%s%s%d%s%s%s%d%s", filesetFilePrefix, separator, t.UnixNano(), separator, suffix, separator, index, fileSuffix)
```

Comment:
Hm we should probably be using `path.Join` here instead, can you update this code to use `path.Join` and the code above too?  Removing the `separator` const would also be good, it's a bad smell to have it as a variable somewhere.